api: regenerate flow.pb.go

Jack-R-lantern opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

- add DebugEventType DBG_L7_LB, DBG_SKIP_POLICY
- regenerate flow.pb.go

Fixes: #27826

gandro commented on bpf/lib/dbg.h:
```diff
@@ -123,6 +123,7 @@ enum {
 				 * arg2: daddr (last 4 bytes for IPv6)
 				 * arg3: proxy port (in host byte order)
 				 */
+	DBG_SKIP_POLICY,
```
Could you explain why this is needed? I don't see it being used in the code anywhere

Jack-R-lantern replied:
@gandro
I understood the sentence in ISSUE-27826 `introducing a new DBG_SKIP_POLICY enum value` as adding a new enum value.
Am I misunderstanding?

gandro replied:
Ah, yes, thanks for the context. The way I understand it, there is more than just the change here needed. One also has to change the datapath to make use of the feature as outlined by Timo (calling it at the end of ipv4_to_endpoint_is_hairpin_flow()").

So I think we should either
a) Also actually use the `DBG_SKIP_POLICY` in the way suggested by Timo (i.e. extend the second commit extend the datapath), or
b) Just do the `DBG_L7_LB` change in this PR (i.e. drop the second commit)

Either are fine with me. If we go for option b), we should remove the "Fixes" reference, since issue #27826 is not complete without implementing a) too.

Jack-R-lantern replied:
@gandro #27798 is merged so I think A is a good choice. I'll fix #27826  in a new PR.

gandro reviewed (changes requested):
Thanks for the PR! I don't fully understand the motivation for the second change

gandro reviewed (approved):
Thanks!

Jack-R-lantern committed:
api: regenerate flow.pb.go

- add DebugEventType DBG_L7_LB, DBG_SKIP_POLICY
- regenerate flow.pb.go
- add bpf/lib/dbg.h Trace types DBG_SKIP_POLICY

Signed-off-by: Jack-R-lantern <tjdfkr2421@gmail.com>

Jack-R-lantern committed:
datapath: add cilium_dbg in tail_ipv4_to_endpoint

This commit add cilium_dbg for locally hairpinned packet

Signed-off-by: Jack-R-lantern <tjdfkr2421@gmail.com>

gandro reviewed (approved):
Hubble related changes look good to me, thanks!
